### PR TITLE
[New Rule] Adding Detection for Stolen Credentials Used to Login to Okta Account After MFA Reset

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/ia-trade-team/issues/170

## Summary
This pull request adds a unique rule that attempts to detect the following TTPs as reported by Okta regarding SCATTERED SPIDER:

_Threat actors appeared to either have a) passwords to privileged user accounts or b) be able to manipulate the delegated authentication flow via Active Directory (AD) prior to calling the IT service desk at a targeted org, requesting a reset of all MFA factors in the target account. In the case of Okta customers, the threat actor targeted users assigned with Super Administrator permissions._

## Additional Information

Screenshot 2023-11-27 at 10 17 36 PM

<details>
<summary>

### Lab Setup

</summary>

1. Okta setup with users and access via console
2. Linux agent deployed to host with Okta integration policy to ingest Okta system logs into the stack
3. Windows Server 2022 deployed
4. Windows agent deployed to host with Windows and Elastic Defend integration policies
5. Sysmon setup on Windows host
6. Procdump installed on Windows host

Screenshot 2023-11-08 at 4 49 43 PM

8. Active Directory (AD) setup and user added to domain forest as administrator

Screenshot 2023-11-08 at 2 41 13 PM

10. From Okta console, Okta directory services agent downloaded and deployed to Windows host
11. AD synced/imported from Windows DC to Okta

Screenshot 2023-11-08 at 2 33 56 PM

Screenshot 2023-11-08 at 2 38 21 PM

13. User assigned an Okta license on import
14. Enabled `Potential Credential Access via Windows Utilities` detection rule

Screenshot 2023-11-09 at 10 24 29 AM

</details>

<details>
<summary>

### Emulation

</summary>

1. While logged into Windows server as user, AV protection disabled
2. From PS console with admin privileges, ran the following: `.\Desktop\Procdump\procdump.exe -ma lsass.exe lsass.dmp`

Screenshot 2023-11-09 at 10 22 30 AM

3. This activity matched the logic of `Potential Credential Access via Windows Utilities` rule which triggered an alert

Screenshot 2023-11-09 at 10 28 06 AM

4. Logged in as a separate user in the Okta console (playing the IT role support), Okta MFA factor was reset for the user whom had just had their LSASS memory dumped

Screenshot 2023-11-08 at 2 57 20 PM

6. In a separate session, I logged into Okta console as the compromised user with my username and stolen credentials and setup the new MFA through my (adversary) device

Screenshot 2023-11-09 at 9 34 22 AM

Screenshot 2023-11-09 at 9 54 39 AM
 
5. I was successfully logged into Okta, had reset MFA to point to my controlled device and since I am a high privilege user, have laterally moved into Okta seeking to make some changes for my benefit :D

Screenshot 2023-11-09 at 9 56 18 AM

</details>

<details>
<summary>

### Detection Logic

</summary>

The logic is relatively straight forward, however there are a couple of things to note. This subquery `any where host.os.type == "windows" and signal.rule.threat.tactic.name == "Credential Access"]` checks for any alert signal where the tactic is for credential access on a windows host. Therefore since we are sequencing with `user.name` we are first checking to see if a some credential access alarm went off first for this user. This is to reduce false-positives dramatically instead of relying solely on MFA factor reset and then a login which is common for Okta users. We are also relying on our detection rule capabilities for credential access, rather that something specific again attempting to take into consideration the information shared by Okta.

We use `any` as the event category in the subqueries of the sequence simply because Okta events are not all given `authentication` (which needs to be an issue to SEI to fix). Also, since we are relying on any credential access alert, it could be `library`, `process`, `file`, etc. therefore is left open.

Our `maxspan` is set to 12 hours which is a wide time window but is on purpose. When the adversary is attempting to first capture credentials on the Windows endpoint, depending on playbook, it could be minutes, hours or even days until they attempt to laterally move into Okta. Therefore without setting too much of a constraint and making the rule inefficient, I settled for 12 hours as the adversary is known to move quickly and adjust proactively during their intrusion efforts.

</details>

### CODE REVIEW

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Review by imays11 — APPROVED** [review]

File: rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml
Comment (w0rk3r) [inline rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml:38]:
```suggestion
- Identify the endpoint for the Credential Access alert for this user by examining the `host.name` and `host.id` fields from the alert document.
```

Comment (w0rk3r) [inline rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml:38]:
```suggestion
- Identify the endpoint for the Credential Access alert for this user by examing the `host.name` and `host.id` fields from the alert document.
- Investigate other alerts associated with the user/host during the past 48 hours.
```

File: rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml
Comment (w0rk3r) [inline rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml:47]:
```suggestion
- If confirmed that the user did not intentionally have their MFA factor reset, deactivate the user account.
```

File: rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml
Comment (w0rk3r) [inline rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml:51]:
```suggestion
- Forensic analysis on the user's endpoint may be required to determine the root cause of the compromise and identify the scope of the compromise.
```

**Review by w0rk3r — APPROVED** [review]
🚀

File: rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml
Comment (terrancedejesus) [inline rules/integrations/okta/persistence_stolen_credentials_used_to_login_to_okta_account_after_mfa_reset.toml:82]:
```suggestion
    [any where event.dataset == "okta.system" and okta.event_type: ("user.session.start", "user.authentication*")]
```

**Comment by terrancedejesus** [comment]
@brokensound77 or @Mikaayenson - Mind taking a look at this one? Need a code owner approval. Thank you!

**Pull request merged**
